rskj-core/src/main/java/org/ethereum/config/net/AbstractNetConfig.java

```diff
@@ -62,4 +71,30 @@ public Constants getCommonConstants() {
         // TODO make a guard wrapper which throws exception if the requested constant differs among configs
         return configs[0].getConstants();
     }
+
+    @Override
+    public Federation getGenesisFederation() {
+        if(genesisFederation == null) {
+            genesisFederation = getCommonConstants().getBridgeConstants().getGenesisFederation();
+        }
+        return genesisFederation;
+    }
+
+    protected final Federation setGenesisFederation(List<BtcECKey> configFederationPublicKeys) {
+
+        BridgeConstants bridgeConstants = getCommonConstants().getBridgeConstants();
+        if(configFederationPublicKeys == null || configFederationPublicKeys.isEmpty()) {
```

Comment:
I don't think we want to assign the `genesisFederation` at all in this case. We just want to keep it null so that the default is returned. So maybe just limit the method to building and setting the `genesisFederation`?

Reply:
i prefer to set it to avoid posible null pointer exceptions in the future